e2e/util_test.go

```diff
@@ -394,6 +394,16 @@ func withWorkingDir(dir *fs.Dir) func(*icmd.Cmd) {
 	}
 }
 
+func withEnv(key, value string) func(*icmd.Cmd) {
+	return func(cmd *icmd.Cmd) {
+		if i := indexSlice(cmd.Env, key+"=", strings.HasPrefix); i > 0 {
+			cmd.Env[i] = key + "=" + value
+		} else {
+			cmd.Env = append(cmd.Env, key+"="+value)
+		}
+	}
+}
+
 type compareFunc func(string) error
```

Comment:
`gotest.tools/v3/icmd` already defines `WithEnv` option;
https://pkg.go.dev/gotest.tools/v3/icmd#WithEnv

Reply:
Yes, but it is not equivalent to what this method is doing. The `icmd.WithEnv` replaces the `Env` field with the arguments. But what I want is to change/set an environment variable (`SHELL` in my case) and leave the rest untouched. The `icmd.WithEnv` is not helpful to do it.